sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/WebSocketsProxyConnectionHandler.java

```diff
@@ -0,0 +1,220 @@
+
+    private InetSocketAddress getProxyAddress() {
+        if (proxyConfiguration != null && proxyConfiguration.isProxyAddressConfigured()) {
+            return (InetSocketAddress) proxyConfiguration.getProxyAddress().address();
+        }
+
+        final URI serviceUri = createURI(amqpHostname, HTTPS_PORT);
+        final ProxySelector proxySelector = ProxySelector.getDefault();
+        if (proxySelector == null) {
+            throw logger.logExceptionAsError(new IllegalStateException(PROXY_SELECTOR_HAS_BEEN_MODIFIED));
+        }
+
+        final List<Proxy> proxies = proxySelector.select(serviceUri);
+        if (!isProxyAddressLegal(proxies)) {
+            throw new IllegalStateException(PROXY_SELECTOR_HAS_BEEN_MODIFIED);
+        }
+
+        final Proxy proxy = proxies.get(0);
+        return (InetSocketAddress) proxy.address();
+    }
+
+    private com.microsoft.azure.proton.transport.proxy.ProxyConfiguration getProtonConfiguration() {
+        final ProxyAuthenticationType type = getProtonAuthType(proxyConfiguration.getAuthentication());
+        final String username = proxyConfiguration.hasUserDefinedCredentials()
+            ? proxyConfiguration.getCredential().getUserName()
+            : null;
+        final String password = proxyConfiguration.hasUserDefinedCredentials()
+            ? new String(proxyConfiguration.getCredential().getPassword())
+            : null;
+
+        return new com.microsoft.azure.proton.transport.proxy.ProxyConfiguration(type,
+            proxyConfiguration.getProxyAddress(), username, password);
+    }
+
+    private ProxyAuthenticationType getProtonAuthType(com.azure.core.amqp.models.ProxyAuthenticationType type) {
+        switch (type) {
+            case DIGEST:
+                return ProxyAuthenticationType.DIGEST;
+            case BASIC:
+                return ProxyAuthenticationType.BASIC;
+            case NONE:
+                return ProxyAuthenticationType.NONE;
+            default:
+                throw logger.logExceptionAsError(new IllegalArgumentException(
+                    "This authentication type is unknown:" + type.name()));
+        }
+    }
+
+    private static URI createURI(final String hostname, final int port) {
+        return URI.create(String.format(Locale.ROOT, HTTPS_URI_FORMAT, hostname, port));
+    }
+
+    /**
+     * This always selects the first proxy in the list instead of going through all the available ones.
+     *
+     * @param proxies List of proxies available.
+     *
+     * @return {@code true} if the first proxy in the list is an HTTP proxy and is an IP address.
+     */
+    private static boolean isProxyAddressLegal(final List<Proxy> proxies) {
+        // we look only at the first proxy in the list
+        // if the proxy can be translated to InetSocketAddress
+        // only then - can we parse it to hostName and Port
+        // which is required by qpid-proton-j library reactor.connectToHost() API
+        return proxies != null
```

Comment:
This is the implementation in Track 1. I wonder if we should search through the list and get any that match rather than the first.

Reply:
Don't have enough context about this. We should leave this as-is to ensure easy migration for clients from track 1 to track 2. Optionally, we can add a configuration to scan all proxies to find a match.